wrapper/src/main/java/software/amazon/jdbc/plugin/staledns/AuroraStaleDnsHelper.java

```diff
@@ -153,4 +161,16 @@ private HostSpec getWriter() {
     }
     return null;
   }
+
+  private boolean isReadOnly(Connection connection, final String query) {
+    try (final Statement statement = connection.createStatement();
+        ResultSet rs = statement.executeQuery(query)) {
+      while (rs.next()) {
+        return rs.getBoolean(AuroraStaleDnsPlugin.IS_READER_COLUMN);
+      }
+    } catch (SQLException e) {
+      return false;
```

Comment:
I'd separate sql syntax exception from others. Syntax exception means that the connection is not to an Aurora cluster and it should re-thrown. Communication-related exceptions probably require a retry. I wouldn't simply return false in such cases.

Reply:
This should probably be an if statement not a while statement